csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.shared.cs

```diff
@@ -589,7 +589,11 @@ public bool EnableProfiling
             {
                 if (!_enableProfiling && value)
                 {
-                    NativeApiStatus.VerifySuccess(NativeMethods.OrtEnableProfiling(handle, NativeMethods.GetPlatformSerializedString(ProfileOutputPathPrefix)));
+                    var pathPrefixBytes = NativeOnnxValueHelper.GetPlatformSerializedString(ProfileOutputPathPrefix);
+                    using (PinnedGCHandle pinnedTrainHandle = new PinnedGCHandle(GCHandle.Alloc(pathPrefixBytes, GCHandleType.Pinned)))
+                    {
+                        NativeApiStatus.VerifySuccess(NativeMethods.OrtEnableProfiling(handle, pathPrefixBytes));
```

Comment:
>athPrefixBytes [](http://example.com/codeflow?start=96&length=14)

Same thing as before. We would need to change C# delegate declaration. Leave this unchanged for now.